### PR TITLE
Add DnD to DataView rows

### PR DESCRIPTION
**Pull request by JAndresHJGlobant**
## What's included?

- Adds drag & drop functionality to the DataView rows using `react-beatiful-dnd`
- Updated DataView documentation including the new `onReorder` props
- Knobs added to storybook to prove this new feature

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,6 +28,7 @@
 		"@types/jest": "^27.4.1",
 		"@types/lodash": "^4.14.179",
 		"@types/react": "^18.0.15",
+		"@types/react-beautiful-dnd": "^13.1.2",
```
Comment (mauriciodeleonc) [inline package.json:31]: I can't add this comment in the yarn.lock file because it's too large, but we shouldn't be committing the yarn.lock file with regular PRs. If you need to install and work with new packages you can without having to install them locally in your computer via docker... for example with this package, you manually add the package and version you need here in the package json and then when you run the `sudo npm run docker` in the putty, it will install that dependency but just in the docker, that way you can work with what you need (it will still probably give you type warnings but just in vscode, which is expected). 

We have a script in place called "sync-state" which basically takes all the yarn.lock from the docker and "copies" it to the local machine but with a new lock version. This script should only be run once we fully merge a PR because if not then we'll get tons of merge conflicts because of all the differences between develop's yarn lock and this (don't worry you don't have to mess with this, this is something I do when merging but just FYI).

TL;DR please re-do this PR without committing the yarn.lock

**Review by mauriciodeleonc — CHANGES_REQUESTED** [review]

File: src/components/DataView/DataViewDisplayList.tsx [hunk 2]
```diff
@@ -65,6 +67,7 @@ function DataViewDisplayList(props: DataViewDisplayListProps) {
 				onCheckAllClick={props.onCheckAllClick}
 				onCheckAllPagesClick={props.onCheckAllPagesClick}
 				onColumnsChange={props.onColumnsChange}
+				onReorder={props?.onReorder}
```
[the remark this answers is not in the document]
Comment (JAndresHJGlobant) [reply src/components/DataView/DataViewDisplayList.tsx:70]: We need it because if reordering is allowed we need to create an extra column that shows the drag-and-drop icon.

File: src/components/DataView/DataViewTHead.tsx [hunk 3]
```diff
@@ -212,6 +213,11 @@ function DataViewTHead(props: DataViewTHeadProps) {
 	return (
 		<StyledWrapper>
 			<tr>
+				{
```
[the remark this answers is not in the document]
Comment (JAndresHJGlobant) [reply src/components/DataView/DataViewTHead.tsx:216]: We need it because if reordering is allowed we need to create an extra column that shows the drag-and-drop icon.

File: src/components/DataView/DataViewTBody.tsx [hunk 4]
```diff
@@ -96,31 +101,60 @@ function DataViewTBody(props: DataViewTBodyProps) {
...
-							onCheckboxClick={onCheckboxClick(i)}
-							checked={props.checked[i]}
-							columns={props.columns}
-						/>
-					)
-				})
-			}
-		</StyledTBody>
-	)
+		<DragDropContext onDragEnd={handleDragEnd}>
```
[the remark this answers is not in the document]
Comment (JAndresHJGlobant) [reply src/components/DataView/DataViewTBody.tsx:133]: In order to use drag and drop, we need to have the part of your React tree that will be able to use drag and drop in wrapped in a ```<DragDropContext />```. In this case, only the rows that are included inside the body of the table need to have access to that Context. You can think of it like a regular React Context API.

[the remark this answers is not in the document]
Comment (JAndresHJGlobant) [reply src/components/DataView/DataViewTBody.tsx:133]: In order to use drag and drop, we need to have the part of your React tree that will be able to use drag and drop in wrapped in a ```<DragDropContext />```. In this case, only the rows that are included inside the body of the table need to have access to that Context. You can think of it like a regular React Context API.

File: src/components/DataView/DataViewTHead.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (JAndresHJGlobant) [reply src/components/DataView/DataViewTHead.tsx:216]: We need it because if reordering is allowed we need to create an extra column that shows the drag-and-drop icon.

**Review by mauriciodeleonc — APPROVED** [review]

File: automation_testing/tests/Components/dataview/save_as.spec.ts [hunk 5]
```diff
@@ -91,11 +91,10 @@ test.describe.parallel("Components - Data View - Save As", () => {
 		await saveAs.viewBtn.click();
 		await saveAs.wait();
 		const editBtn = await saveAs.editBtnByLabel(saveAs_data.saveAsView);
-		// await saveAs.validateSnapshot(editBtn, "save_as_edit_view_btn");
 		expect(await (await saveAs.getViewTypeByLabel(saveAs_data.saveAsView)).textContent()).toContain(saveAs_data.viewNotSharedType);
 		await editBtn.click();
 		await saveAs.saveLabel.fill(saveAs_data.saveAsViewEdit);
-		expect((await saveAs.editCheckbox.isChecked()).valueOf()).toBe(false);
+		await expect(saveAs.saveAsCheckbox).not.toBeChecked();
```
Comment (mauriciodeleonc) [inline automation_testing/tests/Components/dataview/save_as.spec.ts:97]: I'm getting the following error in this line: 
image

**Review by mauriciodeleonc — CHANGES_REQUESTED** [review]

File: automation_testing/tests/Components/dataview/save_as.spec.ts (hunk 5 above)
Comment (torres-nahuel) [reply automation_testing/tests/Components/dataview/save_as.spec.ts]: Fixed

**Review by mauriciodeleonc — APPROVED** [review]
